lib/uffizzi/services/compose_file_service.rb

```diff
@@ -14,8 +14,44 @@ def parse(compose_content, compose_file_path)
       prepare_dependencies(env_files, config_files, compose_file_path)
     end
 
+    def process_env_variables(compose_file_data)
+      compose_file_data.gsub(/\$\{?([?:\-_A-Za-z0-9]+)\}?/) do |var|
+        var_content = var.match(/[?:\-_A-Za-z0-9]+/).to_s
+        fetch_var_value(var_content)
+      end
+    end
+
     private
 
+    def fetch_var_value(var_content)
+      var_name = var_content.match(/^[_A-Za-z0-9]+/).to_s
+      var_value = ENV[var_name]
+      return var_value unless var_value.nil?
+      return fetch_var_default_value(var_content) if var_has_default_value?(var_content)
+
+      if var_has_error_message?(var_content)
+        error_message = fetch_env_error_message(var_content)
+        raise StandardError.new(error_message)
+      end
+      raise StandardError.new("Environment variable #{var_name} doesn't exist")
+    end
```

Comment:
Can we define the error message first based on the condition and raise StandardError one time?